### PR TITLE
docs: shuffle around documentation and examples

### PR DESCRIPTION
**Pull request by adamdecaf**
Fixes: https://github.com/moov-io/ach/issues/353

Demo: https://github.com/adamdecaf/ach/tree/readme-improvements#moov-ioach

### CODE REVIEW

**Comment by adamdecaf** [comment]
Thinking about it our tests are pretty close to examples, but do we want a second set of each SEC type under examples?

File: README.md [hunk 1]
```diff
@@ -46,353 +54,13 @@ ACH is under active development but already in production for multiple companies
 	* Addenda Type Code 18 (IAT Optional)
 	* Addenda Type Code 98 (NOC)
 	* Addenda Type Code 99 (Return)
+-->
 
-## Project Roadmap
-* Additional SEC codes will be added based on library users needs. Please open an issue with a valid test file.
-* Review the project issues for more detailed information
+## HTTP API
```
Comment (adamdecaf) [inline README.md:59]: Should we mention more here? Include a swagger doc just for this project? Refer people to the future `api.moov.io` service?

File: doc.go [hunk 2]
```diff
@@ -2,8 +2,22 @@
...
+//         log.Fatalf("problem opening file: %v", err)
+//     }
+//     file, err := ach.NewReader(fd).Read()
+//     if err != nil {
+//         log.Fatalf("problem parsing ACH file: %v", err)
+//     }
+//     if err := file.Validate(); err != nil {
+//         log.Fatalf("ACH file isn't valid: %v", err)
+//     }
 package ach
```
Comment (adamdecaf) [inline doc.go:23]: This example will show up in the Godoc - should there be more? Links to the table in the README?

**Review by wadearnold — APPROVED** [review]
Thanks for updating this.

**Pull request merged**